src/utilities/xml/XMLValidator.cpp

```diff
@@ -478,7 +478,9 @@ XMLValidator XMLValidator::gbxmlValidator() {
   }
   const bool quiet = true;
   ::openstudio::embedded_files::extractFile(":/xml/resources/GreenBuildingXML_Ver7.03.xsd", openstudio::toString(tmpDir), quiet);
-  return XMLValidator(tmpDir / "GreenBuildingXML_Ver7.03.xsd");
+  auto validator = XMLValidator(tmpDir / "GreenBuildingXML_Ver7.03.xsd");
+  validator.m_tempDir = tmpDir;
```

Comment:
It feels awkward and potentially error prone to need to make this assignment. Maybe we could use a temp directory class like Spawn uses https://github.com/NREL/Spawn/blob/develop/util/temp_directory.hpp and perhaps also move the construction of the temp directory into the construction of the XMLValidator. I understand there might be cases where you don't want XMLValidator to use a temp directory, but perhaps if that is a requirement we could design on overload or something.